SDKAndroid/src/test/java/app/cybrid/sdkandroid/components/bankAccounts/BankAccountsViewModelTest.kt

```diff
@@ -0,0 +1,190 @@
+
+        // -- When
+        viewModel.createWorkflow()
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNotNull(viewModel.workflowJob)
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun test_fetchWorkflow() = runTest {
+
+        // -- Given
+        val dataProvider = prepareClient(JSONMock.JSONMockState.SUCCESS)
+        val viewModel = createViewModel()
+        viewModel.setDataProvider(dataProvider)
+        viewModel.workflowJob = Polling {}
+
+        // -- When
+        viewModel.fetchWorkflow("1234")
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNotNull(viewModel.workflowJob)
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun test_fetchWorkflow_Incomplete() = runTest {
+
+        // -- Given
+        val dataProvider = prepareClient(JSONMock.JSONMockState.EMPTY)
+        val viewModel = createViewModel()
+        viewModel.setDataProvider(dataProvider)
+        viewModel.workflowJob = Polling {}
+
+        // -- When
+        viewModel.fetchWorkflow("1234")
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNotNull(viewModel.workflowJob)
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun test_createExternalBankAccount() = runTest {
+
+        // -- Given
+        val dataProvider = prepareClient(JSONMock.JSONMockState.SUCCESS)
+        val viewModel = createViewModel()
+        viewModel.setDataProvider(dataProvider)
+
+        // -- When
+        viewModel.createExternalBankAccount("", null)
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertEquals(viewModel.UIState.value, BankAccountsView.BankAccountsViewState.DONE)
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun test_checkWorkflowStatus() {
+
+        // -- Given
+        val workflow = WorkflowWithDetailsBankModel(plaidLinkToken = "1234")
+        val viewModel = createViewModel()
+        viewModel.workflowJob = Polling {}
+
+        // -- When
+        viewModel.checkWorkflowStatus(workflow)
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNull(viewModel.workflowJob)
+        Assert.assertEquals(viewModel.latestWorflow, workflow)
+        Assert.assertEquals(viewModel.UIState.value, BankAccountsView.BankAccountsViewState.REQUIRED)
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun test_checkWorkflowStatus_ContinuePolling() {
+
+        // -- Given
+        val workflow = WorkflowWithDetailsBankModel(plaidLinkToken = "")
+        val viewModel = createViewModel()
+        viewModel.workflowJob = Polling {}
+
+        // -- When
+        viewModel.checkWorkflowStatus(workflow)
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNotNull(viewModel.workflowJob)
+        Assert.assertNull(viewModel.latestWorflow)
+        Assert.assertEquals(viewModel.UIState.value, BankAccountsView.BankAccountsViewState.LOADING)
+    }
+}
```

Comment:
These are all positive test cases. You get back many different errors from the various API calls. Check out what you can expect to get back in our OpenAPI spec. It lists them all. We need to test for negative cases as well.

Reply:
Done!